x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/session/EsqlConfiguration.java

```diff
@@ -26,25 +27,30 @@ public class EsqlConfiguration extends Configuration implements Writeable {
 
     private final Locale locale;
 
+    private final TimeValue timeout;
+
     public EsqlConfiguration(
         ZoneId zi,
         Locale locale,
         String username,
         String clusterName,
         QueryPragmas pragmas,
-        int resultTruncationMaxSize
+        int resultTruncationMaxSize,
+        TimeValue timeout
     ) {
         super(zi, username, clusterName);
         this.locale = locale;
         this.pragmas = pragmas;
         this.resultTruncationMaxSize = resultTruncationMaxSize;
+        this.timeout = timeout;
     }
 
     public EsqlConfiguration(StreamInput in) throws IOException {
         super(in.readZoneId(), Instant.ofEpochSecond(in.readVLong(), in.readVInt()), in.readOptionalString(), in.readOptionalString());
-        locale = Locale.forLanguageTag(in.readString());
+        this.locale = Locale.forLanguageTag(in.readString());
         this.pragmas = new QueryPragmas(in);
         this.resultTruncationMaxSize = in.readVInt();
+        this.timeout = in.readOptionalTimeValue();
```

Comment:
Does it need a TransportVersion bump since we are in tech preview and serverless is not out yet?

Reply:
We don't need to bump the version.

Reply:
Serverless is "out" in the sense that it's running in a production environment and serialization errors during upgrades may trip assertions and block a future release. Please don't skip this.

Reply:
We are disabling ES|QL in Serverless now to simplify the developement process.